### ISSUE TITLE
upload function is abnormal

### ISSUE THREAD

**Issue by yueqi77** [e-m-b-a/embark#171]
**Describe the bug**
The firmware upload function is abnormal. The Select file function can be clicked, but no file is selected.

**To Reproduce**
1. EMBArk installation (default mode/docker-mode)
2. Start EMBArk: sudo ./run-server.sh

**Expected behavior**
When the installation is complete, upload functionality should be functional, if database errors do not affect it.

**Screenshots**
Upload firmware
Before selecting a file
![图片](https://github.com/e-m-b-a/embark/assets/32604898/52fafdbc-0708-4bb2-91f1-4f3010cdc079)
After selecting a bin file, nothing happens.
![图片](https://github.com/e-m-b-a/embark/assets/32604898/db3da70b-0a13-4b8f-ae8f-dab90e0e8252)

Upload EMBA-Logs
![图片](https://github.com/e-m-b-a/embark/assets/32604898/01e0e7cf-7764-4dd9-8269-a948e75dbad8)

**Desktop (please complete the following information):**
 - OS: [e.g. Kali Linux 2022.04]

**Additional context**
The database login failure error was reported during the entire installation process, and there were no other err.
![图片](https://github.com/e-m-b-a/embark/assets/32604898/333cdab7-af3c-4955-b62a-4a5f0b96dd61)

In addition, after you submitted the optimization of the dashboard, the embark installation pulled from git always had some problems, such as the errors in the two screenshots below,
I tried to install and process them one by one, but it didn't work. Upgrading or installing incorrect or missing dependencies alone could not solve these errors. However, I pulled git and reinstalled again and again, and each time there were two results. , 1) reported the following two problems, 2) did not report the following two problems
My environment is the same, and the pulled embark resources are also the same, but two results did appear. I don’t know how they came about. This time, only a database error was reported during the installation, but the upload function did not work. More than Firmware upload, it seems that all upload functions are unavailable.
![图片](https://github.com/e-m-b-a/embark/assets/32604898/9dbbf2d3-8d47-44c5-86d1-73714e92d899)
![图片](https://github.com/e-m-b-a/embark/assets/32604898/8377f338-ee71-4237-b292-3b73fe9eeed9)

**Comment by BenediktMKuehne** [comment]
To be able to help you 2 questions:
1. which OS are you using? (EMBArk only supports Ubuntu 22.04 at the moment)
2. is the embeddedanalyzer/emba image up-to-date? (`docker pull embeddedanalyzer/emba:latest`)

Regarding the upload:
is the file upload failing or does the upload button not work - is the POST request happening or not?

**Comment by BenediktMKuehne** [comment]
We also currently are working on a major issue within EMBA in [EMBA - PR913](https://github.com/e-m-b-a/emba/pull/913)
Only once I can merge #172 will EMBArk work normally again

**Comment by yueqi77** [comment]
> To be able to help you 2 questions:
> 
>     1. which OS are you using? (EMBArk only supports Ubuntu 22.04 at the moment)
> 
>     2. is the embeddedanalyzer/emba image up-to-date? (`docker pull embeddedanalyzer/emba:latest`)
> 
> 
> Regarding the upload: is the file upload failing or does the upload button not work - is the POST request happening or not?

Yes, the OS is Ubuntu 22.04.
![image](https://github.com/e-m-b-a/embark/assets/32604898/42281814-d88d-4b4c-a95a-db162422df00)

The tag for the embeddedanalyzer/emba image is latest.
![image](https://github.com/e-m-b-a/embark/assets/32604898/f1125cda-2f4d-4762-b300-72edf4cd5a04)


Upload button doesn't work

**Comment by BenediktMKuehne** [comment]
Good thank you

Are there any javascript errors from inside the browser when you select a file (e.g using the dev-options of the browser)?

**Comment by yueqi77** [comment]
> > Good thank you
> > Are there any javascript errors from inside the browser when you select a file (e.g using the dev-options of the browser)?
> 
>![image](https://github.com/e-m-b-a/embark/assets/32604898/ca02ba72-1b84-4ca2-955e-960558c20b1f)
> 
> ![image](https://private-user-images.githubusercontent.com/32604898/286138979-7f626674-ec12-47a6-95d9-203d49595bc6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExNDkxNjksIm5iZiI6MTcwMTE0ODg2OSwicGF0aCI6Ii8zMjYwNDg5OC8yODYxMzg5NzktN2Y2MjY2NzQtZWMxMi00N2E2LTk1ZDktMjAzZDQ5NTk1YmM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI4VDA1MjEwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgyZDE0MGE3Mzg3ZmFhY2U4NjZkNTAzNmI0MTgwZGU4NDZjYzQzNjU2YzRkNDFlNjM4NTk1NDhkMTk5NWQ2NDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7Y09qa2iSwaR1uBB4WDG-Kcl-tll8JePPzZ7I3a-aBc)

**Comment by BenediktMKuehne** [comment]
This has to be a problem within the fileUpload.js, are there any console errors?

**Comment by yueqi77** [comment]
> This has to be a problem within the fileUpload.js, are there any console errors?

Oh, so that's what it is. There's really an error.
![image](https://github.com/e-m-b-a/embark/assets/32604898/1634d15d-51b4-4ad9-b6b6-0de1e702fe15)


Is it a problem here?
![image](https://github.com/e-m-b-a/embark/assets/32604898/3bdbeef4-15ee-4291-8258-c78fd9ac05c4)


all have
![image](https://github.com/e-m-b-a/embark/assets/32604898/affac3ad-a156-4aaa-bd49-e53b45c482b2)

But I downloaded it separately, not by executing installer.sh. The installer.sh process gets stuck here, and it's hard to get the download to work.
![image](https://github.com/e-m-b-a/embark/assets/32604898/27aa6fe2-8e5e-4c59-9e4b-20ffc624ae56)

**Comment by BenediktMKuehne** [comment]
In here [EMBArk - fileUpload](https://github.com/e-m-b-a/embark/blob/main/embark/static/scripts/fileUpload.js)
Let me check real quick

If the jquery.js shows up in the sources inside the browser everything should be fine there

**Comment by BenediktMKuehne** [comment]
yeah you definitely need all the sources to be downloaded into external/scripts

**Comment by yueqi77** [comment]
> In here [EMBArk - fileUpload](https://github.com/e-m-b-a/embark/blob/main/embark/static/scripts/fileUpload.js) Let me check real quick
> 
> If the jquery.js shows up in the sources inside the browser everything should be fine there

Ah, I don't understand. What am I supposed to do?

**Comment by BenediktMKuehne** [comment]
most likely there is a problem with the jquery.js

I would first check that the `cat embark/static/external/jquery.js` looks something like [jquery-code](https://code.jquery.com/jquery-3.7.1.min.js)

If it does we check what actually ends up in the browser

**Comment by yueqi77** [comment]
> cat embark/static/external/jquery.js

I've zipped up the js and uploaded it so you can see if this is correct
/embark/static/external/scripts/jquery.js
[jquery.zip](https://github.com/e-m-b-a/embark/files/13484486/jquery.zip)

**Comment by BenediktMKuehne** [comment]
Ok looks good

When you reload the upload page, does the jquery file show up and look the same?

**Comment by yueqi77** [comment]
> Ok looks good
> 
> When you reload the upload page, does the jquery file show up and look the same?

Reload the upload page, all the resources are shown to be loaded normally, jquery also responded to the 200ok, but there is a strange phenomenon, the other js open can display the content, only jquery open nothing, in the jquery.js interface click to check the source code, the result is the same nothing to show!
![image](https://github.com/e-m-b-a/embark/assets/32604898/63662361-9d6a-4145-8f9c-597168dad1e0)

**Comment by BenediktMKuehne** [comment]
ok can you check if `/var/www/embark/static/external/scripts/jquery.js` has the expected content?

**Comment by yueqi77** [comment]
> ok can you check if `/var/www/embark/static/external/scripts/jquery.js` has the expected content?

It's empty.
![image](https://github.com/e-m-b-a/embark/assets/32604898/2a22ac52-bd4b-4559-bf4e-c1506b781eee)

**Comment by BenediktMKuehne** [comment]
hmm interesting are you on the current main branch?
the run-server  script should always update newer files between the 2 folders

regardless you may just copy and overwrite /var/www/embark with ./embark since that's the one used for the SGIs

**Comment by yueqi77** [comment]
> hmm interesting are you on the current main branch? the run-server script should always update newer files between the 2 folders
> 
> regardless you may just copy and overwrite /var/www/embark with ./embark since that's the one used for the SGIs

Yes, I just passed a `jquery.js` file from `/embark/static/external/scripts` and restarted `run-server.sh`, but the issue is still not resolved. In the background, the `jquery.js` file in `/var/www/embark/static/external/scripts` still has its content after restarting the server. However, the problem still persists on the interface and the upload functionality still does not work.

![image](https://github.com/e-m-b-a/embark/assets/32604898/8799727f-df1e-4df0-af92-62fb610434e3)
![image](https://github.com/e-m-b-a/embark/assets/32604898/2a06d987-96e6-47ab-8ebc-cd3f23360920)

**Comment by BenediktMKuehne** [comment]
Did you properly refresh and clear the cache?

Is the file still empty (from the web service/browser)? you could check the owner and permissions the file has insed /var/www

**Comment by yueqi77** [comment]
> Did you properly refresh and clear the cache?
> 
> Is the file still empty (from the web service/browser)? you could check the owner and permissions the file has insed /var/www

The system cache is  echo
1, clear the browser cache, turn off and restart the browser
2、Use no trace mode
3、Change a browser that has never been visited before
Still no solution
![image](https://github.com/e-m-b-a/embark/assets/32604898/35832a0c-f697-42c4-a10a-e62f4d5cb47e)

There's something in /var/www.
![image](https://github.com/e-m-b-a/embark/assets/32604898/4513d633-65be-46f3-b1aa-edc26407e803)

Gave attribution to root, restarted run_server.sh, still not resolved
![image](https://github.com/e-m-b-a/embark/assets/32604898/44188b3f-1a4d-422b-abbc-73afeb23d201)

**Comment by BenediktMKuehne** [comment]
Ok this is weird, did you restart the server?

- The apache server serves anything in /var/www/static/* as a static file
- on restart the files from within embark get collected and copied into that folder
- as long as the owner is www-embark and it has read permissions and copies them

question: if you put any random file in that folder (`/var/www/static/external/scripts`) during runtime it should be accessible from the browser. Can you test that?

Note: the WSGI logs are in /var/www/httpd80 and should indicate if there is an issue with serving static files

**Comment by yueqi77** [comment]
> Ok this is weird, did you restart the server?
> 
>     * The apache server serves anything in /var/www/static/* as a static file
> 
>     * on restart the files from within embark get collected and copied into that folder
> 
>     * as long as the owner is www-embark and it has read permissions and copies them
> 
> 
> question: if you put any random file in that folder (`/var/www/static/external/scripts`) during runtime it should be accessible from the browser. Can you test that?
> 
> Note: the WSGI logs are in /var/www/httpd80 and should indicate if there is an issue with serving static files


Yes, on reboot they were copied, so for them the changes were reverted, but in /var/www..... /jquery.js has always had content now

Okay , the test result is that a reboot is required to access the written file

![image](https://github.com/e-m-b-a/embark/assets/32604898/a5aafecf-b1ce-45ce-8ba6-9f85ae6b43f2)
![image](https://github.com/e-m-b-a/embark/assets/32604898/a1143050-f52c-4dfe-992c-188f8eea3c0e)
![image](https://github.com/e-m-b-a/embark/assets/32604898/fb64d134-2ccc-478d-9365-5a2676d47f23)

err_log
The client IP for coding is my local address.
![image](https://github.com/e-m-b-a/embark/assets/32604898/cbad68f9-fb9d-43b3-bd15-64da48ec3337)

**Comment by BenediktMKuehne** [comment]
Yes, outside the developer installation mode all changes only take effect after restarting the run-server.sh script.
All server files have to be owned by that www-embark, it gets changed on startup.

Does the upload work now with the restart?

err_log - looks unrelated

**Comment by yueqi77** [comment]
> Yes, outside the developer installation mode all changes only take effect after restarting the run-server.sh script. All server files have to be owned by that www-embark, it gets changed on startup.
> 
> Does the upload work now with the restart?
> 
> err_log - looks unrelated

Still having problems uploading after reboot (loudly crying face)

**Comment by BenediktMKuehne** [comment]
Are the static files (jquery.js etc) working now?

**Comment by yueqi77** [comment]
> Are the static files (jquery.js etc) working now?

Can't

**Comment by yueqi77** [comment]
Thank God I just made up my mind to reinstall and it ！！！！ Passed in one go ah ah ah ah ah ！！！！！！！ Super excited, well the upload works yay ！！！！！！
However, the execution reported an error (weeping)
Does this mean that there are two directories missing in the scan task directory: /emba_logs/emba.log?
![image](https://github.com/e-m-b-a/embark/assets/32604898/24613147-9341-4e7a-9f25-4cb596b80acf)
![image](https://github.com/e-m-b-a/embark/assets/32604898/e3a128c5-d214-4923-a1c2-740c6e672435)

**Comment by BenediktMKuehne** [comment]
I'm guessing missing www-embark user or missing entry into `/etc/sudoers/`

just re-run these commands
[user creation](https://github.com/e-m-b-a/embark/blob/f40cbc53cf405b71c228384ef411c9e1f344935f/installer.sh#L294)

to test `sudo su www-embark -s /bin/bash` and try to execute emba: `/var/www/emba/emba -d 1`

Context: www-embark has to be able to execute 2 commands as root without password prompt

**Comment by yueqi77** [comment]
> I'm guessing missing www-embark user or missing entry into `/etc/sudoers/`
> 
> just re-run these commands [user creation](https://github.com/e-m-b-a/embark/blob/f40cbc53cf405b71c228384ef411c9e1f344935f/installer.sh#L294)
> 
> to test `sudo su www-embark -s /bin/bash` and try to execute emba: `/var/www/emba/emba -d 1`
> 
> Context: www-embark has to be able to execute 2 commands as root without password prompt

The www-embark user exists, no `/etc/sudoers/` directory
![image](https://github.com/e-m-b-a/embark/assets/32604898/6c9864d9-b321-4b78-a091-1a1b46d35486)
![image](https://github.com/e-m-b-a/embark/assets/32604898/b09d9d37-dd97-44ae-910d-db1e8ae6b9c5)

I re-executed these two commands:
 `echo 'www-embark ALL=(ALL) NOPASSWD: /var/www/emba/emba' | EDITOR='tee -a' visudo`
 `echo 'www-embark ALL=(ALL) NOPASSWD: /bin/pkill' | EDITOR='tee -a' visudo`
And then execute
`sudo su www-embark -s /bin/bash`，smooth execution without password entry
`/var/www/emba/emba -d 1`，I had to add `sudo` to avoid a privilege error and complete the execution without entering a password, but it still gave me an emba error, as shown below.
![image](https://github.com/e-m-b-a/embark/assets/32604898/1ea13b99-2d10-4c3f-a742-5c2db1bf2969)
![image](https://github.com/e-m-b-a/embark/assets/32604898/7d001232-f3cb-4400-a004-86e784bfb394)

**Comment by yueqi77** [comment]
I switched to that directory and it worked!
![image](https://github.com/e-m-b-a/embark/assets/32604898/223ea070-5d78-4cc8-bc6b-8aee67411b5a)

The scan task is starting up fine! Progress has been advancing, there seems to be no problem, the error log is still that my client's message, please see my following screenshot of the scan progress, I crossed out the box that time, what does this time mean ah, the green message of the time seems to be all the time in the future, oh, this should not have an impact on the scan results, right?

![image](https://github.com/e-m-b-a/embark/assets/32604898/f1a4b862-9bea-415a-b5ff-d9f68953d60a)
![image](https://github.com/e-m-b-a/embark/assets/32604898/a5431bd4-d33e-4555-b6e9-b8ef570f063c)

**Comment by yueqi77** [comment]
Oops, this scan went off without a hitch, but without a picture, how do I go about messing with this one?
![image](https://github.com/e-m-b-a/embark/assets/32604898/d7cbd5d2-21f3-4f91-9c4b-41b79d63090e)

**Comment by BenediktMKuehne** [comment]
Any error messages in the browser?
 Looks like a problem on the client-side

**Comment by yueqi77** [comment]
> Any error messages in the browser?
> Looks like a problem on the client-side
Ah, still this jquery.js
![image](https://github.com/e-m-b-a/embark/assets/32604898/74bf25df-2e0e-422a-9a2f-19e0a0df4a90)

**Comment by BenediktMKuehne** [comment]
Likely the charts.js

At this point I would check all your static/external scripts

Code from here:
[Installer-external scipts](https://github.com/e-m-b-a/embark/blob/main/installer.sh#L342)

**Comment by yueqi77** [comment]
> Likely the charts.js
> 
> At this point I would check all your static/external scripts
> 
> Code from here: [Installer-external scipts](https://github.com/e-m-b-a/embark/blob/main/installer.sh#L342)
My God, it's empty！
![image](https://github.com/e-m-b-a/embark/assets/32604898/d9902c6c-1c32-4258-98a7-ea22dbf6af3c)

I downloaded it, oh my god! I finally got it all working,  Installing this was truly like experiencing nine thousand nine hundred eighty-one challenges, it was really difficult (sobbing intensely).
Thank you so much for your patience and guidance, finally I can use the famous emba, oh yeah!
hahahaha I have installed emba before, it is also a very memorable experience, it is the background command to perform a scan, I think it takes a long time, But going through embark and using emba is quick, almost an hour to scan through it， in short, there is a web is very convenient hahahahaha, thank you oh!
![image](https://github.com/e-m-b-a/embark/assets/32604898/38ffab69-df9d-4383-aab9-880123fc8842)
